### ISSUE TITLE
Allow providing kubeconfig from String

### ISSUE THREAD

**Issue by jakublabno** [manusa/helm-java#52]
Hi, as above, it would be nice to use some abstraction for Kubeconfig, not depending only on filesystem Path

Helm.withKubeConfig(String)

**Comment by manusa** [comment]
Hi @jakublabno,

Thanks for your feedback.

Could you please elaborate on your request.

If you're suggesting to allow for the path to be provided as a `String` instead of a `Path` or `File`, then you'd really need to make a good case for that. Personally, I see it as polluting the API and I can't really see or understand a good motive for that (but maybe I'm missing something).

Semantically, for me, `withKubeConfig(String)` means that you provide the `.kube/config` **content** as a `String`.
If this is what you're suggesting, I don't see any reason why we shouldn't provide that. However, it'd be nice to better understand your use case and how you're actually computing or obtaining that `.kube/config` content.

**Comment by jakublabno** [comment]
If you don't want to enhance your API, which is strange since it's common practice, even in official kubernetes java client. you have multiple options to provide config. It's better to provide withKubeConfig(String), it's always easier to convert kubeconfig to string, rather than saving it to the file and provide as File.
Personally, I don't want to store kubeconfig in my project, I store it as base64 in secret and what I have to do now is to crate temporary file which I can later use in withKubeConfig(String)

**Comment by manusa** [comment]
Probably I didn't explain myself correctly.

I'm **happy** to add `withKubeConfig(String)` if this method delivers the functionality to provide the `.kube/config` **content** as a `String`. After your minor elaboration, this is what I now understand you're requesting (as opposed to `withKubeConfig("/home/user/.kube/config")` which is what I won't add unless someone really makes a case for this).

However, you still haven't elaborated on your use case. It's my understanding that you would be running this in a build automation tool and passing the the `.kube/config` retrieving it from a secret. If this is the case, note that `helm-java` is actually leveraging helm libs under the covers, and it's likely that you can pass a lot of information using variables such as: `HELM_KUBEAPISERVER`, `HELM_KUBECAFILE`, `HELM_KUBETOKEN`, and so on).

---

> official kubernetes java client.

I wouldn't use the [Official Kubernetes Java Client](https://github.com/kubernetes-client/java) as an example of API-friendly when you get horrible things such as:
```java
V1PodList list = api.listPodForAllNamespaces(null, null, null, null, null, null, null, null, null);
```

Especially when you can use the [Fabric8 Kubernetes Client](https://github.com/fabric8io/kubernetes-client) that provides completely fluent interfaces :wink: 

```java
PodList podList = client.pods().inAnyNamespace().list();
```

**Comment by Jean-Jacques-Nuoan** [comment]
> Probably I didn't explain myself correctly.
> 
> I'm **happy** to add `withKubeConfig(String)` if this method delivers the functionality to provide the `.kube/config` **content** as a `String`. 

May I ask when this idea will be implemented

**Comment by miriSch** [comment]
For me, it also would be very helpful to be able to configure helm operations with kube config file contents like it is also possible for the Fabric8 Kubernetes Client: https://github.com/fabric8io/kubernetes-client/blob/faedc39b0dc49deee5bb962068eecf61128cf74b/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/Config.java#L782

Our use case is to build an application which can manage installations in multiple kubernetes clusters from a single cluster using Fabric8 Kubernetes Client and helm.

**Comment by manusa** [comment]
I've just published v0.0.15 which includes the feature requested in this issue:

- https://repo1.maven.org/maven2/com/marcnuri/helm-java/helm-java/0.0.15/
- https://github.com/manusa/helm-java/releases/tag/0.0.15

Please, share any feedback you may have, it is much appreciated :pray:
